src/mmap.rs

```diff
@@ -307,6 +324,48 @@ impl GuestMemoryMmap {
             regions.push(GuestRegionMmap {
                 mapping,
                 guest_base: range.0,
+                file: None,
+                offset: None,
+            });
+        }
+
+        Ok(Self {
+            regions: Arc::new(regions),
+        })
+    }
+
+    /// Creates a container and allocates shared memory backed by a file for
+    /// guest memory regions. Valid memory regions are specified as a Vec of
+    /// (Address, Size, File, Offset) tuples sorted by Address.
+    pub fn new_backed_by_file(
```

Comment:
I guess this is mainly a forward looking comment, in the sense it doesn't necessarily fit in the scope of this PR, but it feels a bit redundant to have three constructors for  `GuestMemoryMmap`, which also share quite a bit of logic (like the overlapping regions check). I think it would be neat to only keep `from_regions`, and then implement some helper logic in `mmap_unix.rs` (and the Windows one I guess) to simplify creating `Vec`s of regions (file backed or not), which can then be passed to `from_regions`. Just throwing this out there, what do you ppl think?

Reply:
Yes I agree with this one, but let's do it in 2 steps if you don't mind. Let's first introduce the new possibility to create memory regions backed by a file, and then we can try to provide all features through a factorized function.

Reply:
Sure

Reply:
Let's add the `FileRange` struct, and make this take a `&[(GuestAddress, FileRange)]`.

Reply:
@alexandruag will include it in its PR.